server/manageuser.php

```diff
@@ -0,0 +1,178 @@
+		$tbody.= "</tr>";
+	}
+	$tbody.= "</tbody>";
+	$thead.= "</tr></thead>";
+
+	$response['table'] = $thead.$tbody;
+
+	print json_encode($response);
+}
+
+// Returns string of html <option> elments with all permissions
+function getUserPermissionOptionList($userId){
+	GLOBAL $DB_CONN;
+
+	$stmt = $DB_CONN->prepare("SELECT 
+		permissionId,
+		name,
+        description,
+        lookupName,
+        (SELECT userPermissionId FROM UserPermission WHERE userId = ? AND permissionId = p.permissionId) as userPermissionId
+	FROM Permission p");
+
+	$stmt->execute(array($userId));
+
+	$options = array();
+
+	$count = 0;
+	while($row = $stmt->fetch(PDO::FETCH_ASSOC)){
+		if($row['userPermissionId']){
+			$selected = 'selected=true';
+			$data = "data-userPermissionId='".$row['userPermissionId']."'";
+			$count++;
+		}else{
+			$selected = '';
+			$data = '';
+		}
+		$options[] = "<option $data value=".$row['permissionId']." $selected>".$row['name']."</option>";
+	}
+
+	return array(
+		'options' => $options,
+		'hasPermissionCount' => $count
+	);
+}
+
+function updateUserPermissions($data){
+	GLOBAL $DB_CONN;
+	GLOBAL $USER;
+
+
+	$response = array();
+	$response['success'] = true;
+
+	if($data['userId'] == $USER->getUserId()){
+		$stmt = $DB_CONN->prepare("SELECT lookupName 
+			FROM vita.permission
+				INNER JOIN vita.userpermission ON permission.permissionId = userpermission.permissionId
+			WHERE userPermissionId = ?");
+
+		foreach ($data['removePermissionArr'] as $userPermissionId) {
+			$stmt->execute(array($userPermissionId));
+			$lookupName = $stmt->fetch(PDO::FETCH_ASSOC)['lookupName'];
+
+			if($lookupName == 'edit_user_permission'){
+				$firstName = $USER->getUserDetails()['firstName'];
+
+				$response['success'] = false;
+				$response['error'] = "I'm sorry, $firstName, I'm afraid I can't do that.";
+				print json_encode($response);
+				die();
+			}
+		}
+	}
+
+	$DB_CONN->beginTransaction();
+
+	if(isset($data['removePermissionArr'])){
+		$stmt = $DB_CONN->prepare("DELETE FROM UserPermission WHERE userPermissionId = ?");
+
+		foreach ($data['removePermissionArr'] as $userPermissionId) {
+			$stmt->execute(array($userPermissionId));
+		}
+	}
+
+	if(isset($data['addPermissionArr'])){
+		$stmt = $DB_CONN->prepare("INSERT INTO UserPermission 
+				(userId, permissionId, createdBy)
+			VALUES 
+				(?, ?, ?)");
+
+		foreach ($data['addPermissionArr'] as $permissionId) {
+			$stmt->execute(array(
+				$data['userId'],
+				$permissionId, 
+				$USER->getUserId()
+			));
+		}
+	}
+
+	$DB_CONN->commit();
+
+	print json_encode($response);
+}
+
+function addUser($data){
+	GLOBAL $DB_CONN;
+
+	$response = array();
+	$response['success'] = true;
+
+	$stmt = $DB_CONN->prepare("INSERT INTO User 
+			(firstName, lastName, email, phoneNumber, preparesTaxes)
+		VALUES 
+			(?, ?, ?, ?, ?)");
+
+	$res = $stmt->execute(array(
+		$data['firstName'],
+		$data['lastName'],
+		$data['email'],
+		$data['phone'],
+		$data['prepareTaxes']
```

Comment:
I think the issue with prepares taxes is right here! You forgot the "s" on "prepares". And also, do you think it's a good idea to transform the `$data['preparesTaxes']` to 0 or 1?

Reply:
Adding a conversion client side fixed it. My bad on missing this. Thanks for catching it!